src/libsyntax/parse/parser.rs

```diff
@@ -560,7 +561,9 @@ impl<'a> Parser<'a> {
     }
 
     pub fn parse_ident(&mut self) -> PResult<'a, ast::Ident> {
-        self.check_strict_keywords();
+        if !self.restrictions.contains(Restrictions::ALLOW_MODULE_PATHS) {
+            self.check_strict_keywords();
```

Comment:
Lol.
I'm pretty sure, naked `self` and `super` should *always* be parseable as paths/idents and resolved later as usual.
There are even a couple of issued about this - https://github.com/rust-lang/rust/issues/29036, https://github.com/rust-lang/rust/issues/27899.
But not this time, I guess.